source/extensions/omni.isaac.lab/omni/isaac/lab/scene/interactive_scene.py

```diff
@@ -496,7 +501,138 @@ def _add_entities_from_cfg(self):
+        physx_scene.CreateInvertCollisionGroupFilterAttr().Set(True)
+
+        # Create a scope to store collision groups
+        UsdGeom.Scope.Define(self.stage, "/Collisions")
+
+        for group_name in self._collision_groups:
+            if group_name == "global":
+                continue
+            UsdGeom.Scope.Define(self.stage, f"/Collisions/{group_name}")
+
+        with Sdf.ChangeBlock():
+            # create the global group
+            if self._collision_groups["global"]:
+                # specify the prim path for the group
+                global_collision_group_path = "/Collisions/global_group"
+
+                # add collision group prim
+                global_collision_group = Sdf.PrimSpec(
+                    self.stage.GetRootLayer().GetPrimAtPath("/Collisions"),
+                    "global_group",
+                    Sdf.SpecifierDef,
+                    "PhysicsCollisionGroup",
+                )
+                # prepend collision API schema
+                global_collision_group.SetInfo(
+                    Usd.Tokens.apiSchemas, Sdf.TokenListOp.Create({"CollectionAPI:colliders"})
+                )
+
+                # expansion rule
+                expansion_rule = Sdf.AttributeSpec(
+                    global_collision_group,
+                    "collection:colliders:expansionRule",
+                    Sdf.ValueTypeNames.Token,
+                    Sdf.VariabilityUniform,
+                )
+                expansion_rule.default = "expandPrims"
+
+                # includes rel
+                global_includes_rel = Sdf.RelationshipSpec(
+                    global_collision_group, "collection:colliders:includes", False
+                )
+                for prim_path in self._collision_groups["global"]:
+                    global_includes_rel.targetPathList.Append(prim_path)
+
+                # filteredGroups rel
+                global_filtered_groups = Sdf.RelationshipSpec(global_collision_group, "physics:filteredGroups", False)
+                # We are using inverted collision group filtering, which means objects by default don't collide across
+                # groups. We need to add this group as a filtered group, so that objects within this group collide with
+                # each other.
+                global_filtered_groups.targetPathList.Append(global_collision_group_path)
+
+            # create local groups
+            for group_name, prim_path_exprs in self._collision_groups.items():
+                if group_name == "global":
+                    continue
+
+                # Check if we have any sub-groups
+                if len(self._collision_groups) > 2:
+                    # find matching prim paths
+                    prim_paths = []
+                    for prim_path_expr in prim_path_exprs:
+                        prim_paths += sim_utils.find_matching_prim_paths(prim_path_expr)
+                else:
+                    prim_paths = self.env_prim_paths
```

Comment:
There's a new feature from physics that will let us do the environment collision filtering automatically through physics replication. we should probably think about how to incorporate this with the collision group logic.